contracts/HeapOrdering.sol

```diff
@@ -168,18 +169,19 @@ library HeapOrdering {
         HeapArray storage _heap,
         address _id,
         uint96 _value,
-        uint256 _maxSortedUsers
+        uint128 _maxSortedUsers
     ) private {
         // `_heap` cannot contain the 0 address.
         if (_id == address(0)) revert AddressIsZero();
 
         // Put the account at the end of accounts.
-        uint256 accountsLength = _heap.accounts.length;
-        _heap.accounts.push(Account(_id, _value));
+        uint128 accountsLength = _heap.arrayLength;
+        _heap.accounts[accountsLength] = Account(_id, _value);
+        _heap.arrayLength = accountsLength + 1;
```

Comment:
```suggestion
        uint128 accountsLength = _heap.arrayLength++; // Increase _heap.arrayLength while returning the former value.
        _heap.accounts[accountsLength] = Account(_id, _value);
```

Reply:
this actually increases the gas a little bit 😞

Reply:
woww really? Note sure to get why....

Btw gas is tested at 200 optimizer runs, we might need to test at an higher number of runs.